crates/ra_project_model/src/lib.rs

```diff
@@ -265,7 +266,11 @@ impl ProjectWorkspace {
                             json_project::Edition::Edition2018 => Edition::Edition2018,
                         };
                         let cfg_options = {
-                            let mut opts = CfgOptions::default();
+                            let target = krate.target.as_deref();
+                            let mut opts = default_cfg_map
+                                .entry(target)
+                                .or_insert_with(|| get_rustc_cfg_options(target))
```

Comment:
Hm, my gut feeling is we shouldn't use `get_rustc_cfg_options` here at all, and instead just use what is provided to us. 

This allows the users to have full control over the cfg flags. 

OTOH, the users would essentially need to replicate `get_rustc_cfg_options` logic, so it might makes sense to add this as a convenience thing...

I guess, the convenient and flexible logic should be:

* if target is set, mix rustc's cfgs in
* otherwise, just use passed in cfgs, *do not* fallback to the default target.

Unrelated, but I also look at the `target: Option<&str>,` and think that it doesn't really make sense? Even for standard Cargo projects, you often have two targets active:

* host target for build scripts and proc macros
* target target for the final binary

Reply:
In an internal thread, someone brought up librarification. If we could get the list of cfgs enabled by target, then the logic could be built into RA without needing to rely on rustc being on the PATH. Our thought was that the target specific cfgs change so rarely it might make sense to build it into RA. Not sure if @woody77 might have anything else to add here. We did also bring up host vs target targets. I think that's something I should include in this design.

Reply:
Ok, either way, I think we should be able to tweak this down the line. 

I am also fine either way on

>otherwise, just use passed in cfgs, do not fallback to the default target.

We can, or can not do this. 

so, 

bors d+

Reply:
ping @Nashenas88

Reply:
I'll try addressing this tomorrow. The merge conflicts require looking into this a bit more since they directly affect the code from this PR, didn't have time earlier in the week.